packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/sanitizeOverFlowingChildren.tsx

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Span } from '../../../../types/trace';
+
+// This function truncates/drops the overflowing child spans
```

Comment:
circular definition. 
  * explain what overflowing mean
  * are childOf and followsFrom children treated the same?
  * what is the scope here? `spans: Span[]` - is this the whole trace?

Reply:
https://github.com/jaegertracing/jaeger-ui/blob/eb28e9ebd074d1640d61e41e1e32cacecf462120/packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/sanitizeOverFlowingChildren.tsx#L17-L22
Present it treats childOf and followsFrom children as same. But getChildOf spans removes the followsFrom children after sanitizing https://github.com/jaegertracing/jaeger-ui/blob/e8e678524bfb2040d27934825260eb78d54086ef/packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/getChildOfSpans.tsx#L18
Yes `spans: Span[ ]` is the whole trace

Reply:
https://github.com/jaegertracing/jaeger-ui/blob/36ce16b582d0a54d300014e9e228dc7cff21e772/packages/jaeger-ui/src/components/TracePage/CriticalPath/utils/sanitizeOverFlowingChildren.tsx#L17-L26
Is this function definition fine for sanitizeOverFlowingChildren function??